app/screens/edit_profile/edit_profile.tsx

```diff
@@ -170,15 +216,47 @@ const EditProfile = ({
         enableSaveButton(true);
     }, [enableSaveButton]);
 
-    const onUpdateField = useCallback((fieldKey: string, name: string) => {
+    const onUpdateField = useCallback((fieldKey: string, value: string) => {
         const update = {...userInfo};
-        update[fieldKey] = name;
+        if (fieldKey.startsWith('customAttributes.')) {
+            const attrKey = fieldKey.split('.')[1];
+            update.customAttributes = {...update.customAttributes, [attrKey]: {id: attrKey, name: userInfo.customAttributes[attrKey].name, value}};
+        } else {
+            switch (fieldKey) {
+            // typescript doesn't like to do update[fieldkey] as it might containg a customAttribute case
+                case 'email':
+                    update.email = value;
+                    break;
+                case 'firstName':
+                    update.firstName = value;
+                    break;
+                case 'lastName':
+                    update.lastName = value;
+                    break;
+                case 'nickname':
+                    update.nickname = value;
+                    break;
+                case 'position':
+                    update.position = value;
+                    break;
+                case 'username':
+                    update.username = value;
+                    break;
+            }
+        }
         setUserInfo(update);
 
-        // @ts-expect-error access object property by string key
-        const currentValue = currentUser[fieldKey];
-        const didChange = currentValue !== name;
-        hasUpdateUserInfo.current = currentValue !== name;
+        let didChange = false;
+        if (fieldKey.startsWith('customAttributes.')) {
+            const attrKey = fieldKey.split('.')[1];
```

Comment:
Is there any possibility the custom attribute contains dots? (like customAttributes.location.city)

If that is an option, you may have to rethink this. There are several solutions here, but not sure which one is the best: https://stackoverflow.com/questions/4607745/split-string-only-on-first-instance-of-specified-character